### ISSUE TITLE
Conditional Row Styles create a new class instead of being inline style

### ISSUE THREAD

**Issue by MedoHaleem** [jbetancur/react-data-table-component#772]
## Issue Check list
- [ x] Agree to the [Code of Conduct](https://github.com/jbetancur/react-data-table-component/blob/master/CODE-OF-CONDUCT.md)
- [x ] Read the README
- [x ] You are using React 16.8.0+
- [x ] You installed `styled-components`
- [x ] Include relevant code or preferably a [code sandbox](https://codesandbox.io/embed/react-data-table-sandbox-ccyuu
)

## Describe the bug
Conditional Row Styles create a new class instead of being inline change and we get the following message:
```
Over 200 classes were generated for component styled.div with the id of "sc-jrAGrp".
Consider using the attrs method, together with a style object for frequently changed styles.
Example:
  const Component = styled.div.attrs(props => ({
    style: {
      background: props.background,
    },
  }))`width: 100%;`
```

  <Component />
## To Reproduce
Simply add Conditional row style that depends on row value to create css style and update the row frequently (for example the table updates when new data come from websocket).

```
    {
      when: _row => true,
      style: row => ({
        background: `linear-gradient(90deg, rgba(255,88,87,0.1) ${calculateBackgroundBar(row, BidsData)}, rgba(0,0,0,1) ${calculateBackgroundBar(row, BidsData)})`,
        color: 'white'
      }),
    }

```
## Expected behavior
styled-components shouldn't create over 200 classes

## Versions (please complete the following information)
 - React 17.0.1
 - Styled Components 5.2.1
 - OS:  Windows 10 2004
 - Browser: Chrome

**Comment by jbetancur** [comment]
Nice catch. I'll add this to the next branch as a performance enhancement.

**Comment by jbetancur** [comment]
fixed #777 alpha-6
